File: 0868-binary-gap/0868-binary-gap.cpp
class Solution {
public:
    int binaryGap(int n) {
        vector<int> vec;
        while(n > 0){
            vec.push_back(n % 2);
            n /= 2;
        }
        reverse(vec.begin(), vec.end());
        int ans = 0, cnt = 0;
        for(int i = 1; i < vec.size(); i++){
            if(vec[i] == 0){
                cnt++;
            }else{
                ans = max(ans, cnt + 1);
                cnt = 0;
            }
        }
        return ans;
    }
};

File: 0868-binary-gap/0868-binary-gap.cs
public class Solution {
    public int BinaryGap(int n) {
        //  https://leetcode.com/problems/binary-gap/solutions/3181929/beats-98-easy-solution-please-upvote
        int last = -1, ans = 0;
        for(int i = 0; i < 32; i++){
            if(((n >> i) & 1) > 0){
                if(last >= 0){
                    ans = Math.Max(ans, i - last);
                }
                last = i;
            }
        }

        return ans;
    }
}